File: SmackDown2019/PreElims/CHQUEENS.cpp
#include <bits/stdc++.h>
using namespace std;


int min(int a, int b){
  if (a<b) return a;
  return b;
}

int givesquare(int a, int b, int x, int y, int n, int m){
  int ans=0;

  if (a==x && b==y){
    return 0;
  }

  if (b==y && x<a){
    ans += a-x-1;
  }
  else{
    ans += a-1;
  }

  if (b==y && x>a){
    ans += x-a-1;
  }
  else{
    ans += n-a;
  }


  if (a==x && y>b){
    ans += y-b-1;
  }
  else{
    ans += m-b;
  }

  if (a==x && y<b){
    ans += b-y-1;
  }
  else{
    ans += b-1;
  }

  if ((x-y == a-b) && x<a){
    ans += a-x-1;
  }
  else{
    ans += min(a-1,b-1);
  }

  if ((x-y == a-b) && x>a){
    ans += x-a-1;
  }else{
    ans += min(n-a,m-b);
  }

  if ((x+y == a+b) && y>b){
    ans += y-b-1;
  }else{
    ans += min(a-1,m-b);
  }

  if ((x+y == a+b) && y<b){
    ans += b-y-1;
  }else{
    ans += min(n-a,b-1);
  }

  return ans;
}

int main(){
  int t;
  cin>>t;
  int arr[t][4];
  for(int i=0;i<t;i++){
    cin>>arr[i][0];
    cin>>arr[i][1];
    cin>>arr[i][2];
    cin>>arr[i][3];
  }

  int n,m,x,y,asdf,ans=0;
  for(int i=0;i<t;i++){
    ans = 0;
    n = arr[i][0];
    m = arr[i][1];
    x = arr[i][2];
    y = arr[i][3];
    asdf = (n*m)-2;
    

    for(int a=1;a<=n;a++){
      for(int b=1;b<=m;b++){
        if(a==x && b==y) continue;
        ans += asdf-givesquare(a,b,x,y,n,m);
      }
    }
    cout<<ans<<endl;
  }
}

File: SmackDown2019/PreElims/CHQUEENS.rb
def debug a
#  puts a
end

def debug2 a
#  puts a
end

def givesquare(a,b,x,y,n,m)

#  debug2 "#{a} #{b} #{x} #{y} #{n} #{m}"

  if a==x and b==y
    return 0
  end

  ans = 0
  #a,b is my position
  #x,y is his position

  #look up
  if b==y and x<a
    ans += a-x-1
  else
    ans += a-1
  end
#  debug "UP:" + ans.to_s

  #look down
  if b==y and x>a
    ans += x-a-1
  else
    ans += n-a
  end
#  debug "DOWN:" + ans.to_s
  #look right
  if a==x and y>b
    ans += y-b-1
  else
    ans += m-b
  end
#  debug "RIGHT:" + ans.to_s
  #look left
  if a==x and y<b
    ans += b-y-1
  else
    ans += b-1
  end
#  debug "LEFT:" + ans.to_s
  #look upleft
  if (x-y == a-b) and x<a
    ans += a-x-1
  else
    ans += [a-1,b-1].min
  end
#  debug "UPLEFT:" + ans.to_s
  #look downright
  if (x-y == a-b) and x>a
    ans += x-a-1
  else
    ans += [n-a,m-b].min
  end
#  debug "DOWNRIGHT:" + ans.to_s
  #look upright
  if (x+y == a+b) and y>b
    ans += y-b-1
  else
    ans += [a-1,m-b].min
  end
#  debug "UPRIGHT:" + ans.to_s
  #look downleft
  if (x+y == a+b) and y<b
    ans += b-y-1
  else
    ans += [n-a,b-1].min
  end
#  debug "DOWNLEFT:" + ans.to_s
  return ans
end

t = gets.to_i
arr = []
t.times {arr.push gets.chomp.split(" ").map(&:to_i)}

arr.each do |nmxy|
  n = nmxy[0]
  m = nmxy[1]
  x = nmxy[2]
  y = nmxy[3]

  ans = 0
  (1..n).each do |a|
    (1..m).each do |b|
      next if a==x and b==y
      ans += (n*m)-givesquare(a,b,x,y,n,m)-2
#      puts "#{a} #{b} #{givesquare(a,b,x,y,n,m)}"
    end
  end
  puts ans
end
